src/models/request.ts

```diff
@@ -15,6 +28,37 @@ export class Request {
   pinned: boolean
   createdAt: Date
   updatedAt: Date
+  timestamp: Date
+  origin?: string
+
+  constructor(params: Partial<Request> = {}) {
+    // TODO Proper input types
+    this.id = params.id
+    this.status = params.status
+    this.cid = params.cid
+    this.streamId = params.streamId
+    this.message = params.message
+    this.pinned = params.pinned
+    this.createdAt = params.createdAt
+    this.updatedAt = params.updatedAt
+    this.timestamp = params.timestamp
+    this.origin = params.origin
+  }
+
+  toDB(): IDBRequest {
```

Comment:
Can we add a comment what this is doing and why it's necessary, with a big disclaimer making clear that you *have* to use this before inserting a request into the database or else your data will be silently corrupted?